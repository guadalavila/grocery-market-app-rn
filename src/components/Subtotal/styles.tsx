import React from 'react';
import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { FontSize, Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: Colors.white,
        borderRadius: Sizes.S,
        paddingHorizontal: Sizes.M,
        padding: Sizes.S,
        marginTop: Sizes.S,
        marginBottom: Sizes.L,
        marginHorizontal: Sizes.L,
    },
    title: {
        fontSize: FontSize.L,
        color: Colors.black,
        fontWeight: '600',
    },
    total: {
        color: Colors.black,
        fontWeight: 'bold',
        fontSize: FontSize.XL,
    },
    subtitle: {
        fontWeight: '600',
    },
});
