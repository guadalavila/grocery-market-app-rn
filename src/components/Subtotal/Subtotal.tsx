import React from 'react';
import { Text, View } from 'react-native';
import { Colors } from '../../utils/Colors';
import { GlobalStyles, Sizes } from '../../utils/GlobalStyles';
import { Strings } from '../../utils/Strings';
import Box from '../Box/Box';
import Divider from '../Divider/Divider';
import { styles } from './styles';

interface ISubtotalProps {
    total: number;
}

const Subtotal = ({ total }: ISubtotalProps) => {
    return (
        <View style={styles.container}>
            <Box mb={Sizes.S} flexDirection='row' justifyContent='space-between'>
                <Text style={styles.subtitle}>{Strings.subtotal}</Text>
                <Text testID='subtotal'>${total}.00</Text>
            </Box>
            <Box mb={Sizes.S} flexDirection='row' justifyContent='space-between'>
                <Text style={styles.subtitle}>{Strings.delivery}</Text>
                <Text>$10.00</Text>
            </Box>
            <Divider height={0.2} color={Colors.greyLight} marginHorizontal={0} />
            <Box mb={Sizes.S} mt={Sizes.S} flexDirection={'row'} justifyContent={'space-between'}>
                <Text style={styles.title}>{Strings.total}</Text>
                <Text testID='total' style={[GlobalStyles.center, styles.total]}>
                    ${total + 10}.00
                </Text>
            </Box>
        </View>
    );
};

export default Subtotal;
