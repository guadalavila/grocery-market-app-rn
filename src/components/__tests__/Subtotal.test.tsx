import { render, screen } from '@testing-library/react-native';
import React from 'react';
import Subtotal from '../Subtotal/Subtotal';

describe('test Subtotal component', () => {
    test('should match snapshot', () => {
        const component = render(<Subtotal total={100} />);
        expect(component).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<Subtotal total={100} />);
        expect(screen).toBeTruthy();
    });

    test('should render texts', () => {
        render(<Subtotal total={100} />);
        expect(screen.getByText('Subtotal')).toBeTruthy();
        expect(screen.getByText('Total')).toBeTruthy();
        expect(screen.getByTestId('subtotal').children[1]).toContain('100');
        expect(screen.getByTestId('total').children[1]).toContain('110');
    });
});
