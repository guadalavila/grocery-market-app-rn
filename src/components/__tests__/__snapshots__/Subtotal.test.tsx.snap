// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Subtotal component should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "marginBottom": 24,
      "marginHorizontal": 24,
      "marginTop": 8,
      "padding": 8,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": undefined,
        "backgroundColor": "transparent",
        "borderRadius": undefined,
        "display": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "space-between",
        "margin": 0,
        "marginBottom": 8,
        "marginHorizontal": 0,
        "marginTop": 0,
        "marginVertical": 0,
        "padding": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "paddingVertical": 0,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "600",
        }
      }
    >
      Subtotal
    </Text>
    <Text
      testID="subtotal"
    >
      $
      100
      .00
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": undefined,
        "backgroundColor": "transparent",
        "borderRadius": undefined,
        "display": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "space-between",
        "margin": 0,
        "marginBottom": 8,
        "marginHorizontal": 0,
        "marginTop": 0,
        "marginVertical": 0,
        "padding": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "paddingVertical": 0,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "600",
        }
      }
    >
      Delivery / Extra
    </Text>
    <Text>
      $10.00
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {},
        Object {
          "backgroundColor": "#C5C5C5",
          "height": 0.2,
          "marginHorizontal": 0,
          "marginVertical": 16,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignSelf": undefined,
        "backgroundColor": "transparent",
        "borderRadius": undefined,
        "display": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "space-between",
        "margin": 0,
        "marginBottom": 8,
        "marginHorizontal": 0,
        "marginTop": 8,
        "marginVertical": 0,
        "padding": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "paddingVertical": 0,
        "width": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      Total
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "#000000",
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      }
      testID="total"
    >
      $
      110
      .00
    </Text>
  </View>
</View>
`;
